### DISCUSSION TITLE
Emulated elements and visibility

### DISCUSSION THREAD

**Discussion by maoudia-via** [Consensys/gnark#384] [Q&A] [answered]
The emulated example on the develop branch only shows default visibility on the circuit definition.
When trying to define a circuit with different visibility for some variables I see the following error:

`Y_limbs (secret) has a parent with different visibility attribute`

This is my Circuit definition (at least the struct)

```
type MyCircuit struct {
	X             emulated.Element[emulated.Goldilocks] `gnark:"X,"`
	Y             emulated.Element[emulated.Goldilocks] `gnark:"Y,public"`
}
```

Is there a way to define an emulated element with public visibility ?

**Comment by ivokub** [comment, marked as answer]
This is most likely a bug. I'll try to understand and fix it.

**Reply by maoudia-via** [reply]
Thank you @ivokub  I can open an issue and provide a sample if that's helpful.

**Reply by ivokub** [reply]
Thanks. I already created the issue https://github.com/ConsenSys/gnark/issues/385. I already understand where the problem is, but will try to figure out if this wouldn't break anything.

**Reply by ivokub** [reply]
Sorry for the delay. I was traveling last week. Could you try with this PR https://github.com/ConsenSys/gnark/pull/387?

**Reply by maoudia-via** [reply]
Thank you ! I will try it this week and get back to you.

**Reply by maoudia-via** [reply]
I can confirm my scenario/test is fixed by with PR #387. 
Thank you for the prompt feedback and fix !

**Reply by ivokub** [reply]
Great! Thanks for testing out. The field emulation feature is still very new and we need to figure out a clean way to provide it to the users. Any feedback is greatly appreciated!
